cedar-policy/src/api.rs

```diff
@@ -3900,6 +3918,14 @@ mod schema_based_parsing_tests {
     use cool_asserts::assert_matches;
     use serde_json::json;
 
+    #[test]
+    fn accessing_unspecified_entity_returns_none() {
+        let c = Context::empty();
+        let request: Request = Request::new(None, None, None, c);
+        let p = request.principal();
+        assert!(p.is_none());
```

Comment:
just for completeness we could check `request.action()` and `request.resource()` here as well